src/orbits/keplerian.jl

```diff
@@ -0,0 +1,532 @@
+using AstroLib: trueanom, kepler_solver
+using PhysicalConstants
+using Unitful, UnitfulAstro
+using KeywordCalls
+using Rotations
+
+# Domain specific unit conversions / Constants
+const G_nom = 2942.2062175044193 # Rsun^3/Msun/d^2
+const G_unit = PhysicalConstants.CODATA2018.G
+const MsunRsun_to_gcc = (1.0u"Msun/Rsun^3" |> u"g/cm^3").val
+
+"""
+    KeplerianOrbit(; kwargs...)
+
+Keplerian orbit parameterized by the basic observables of a transiting 2-body system.
+
+# Parameters
+* `period`/`P` -- The orbital period of the planet [d].
+* `t_0`/`t₀` -- The midpoint time of the reference transit [d].
+* `t_p`/`tₚ` -- The time of periastron [d].
+* `duration`/`τ` -- The transit duration [d].
+* `a` -- The semi-major axis [R⊙].
+* `R_planet`/`Rₚ` -- The radius of the planet [R⊙].
+* `R_star`/`Rₛ` -- The radius of the star [R⊙].
+* `rho_star`/`ρₛ` -- The spherical star density [M⊙/R⊙³].
+* `RpRs` -- The ratio of the planet radius to star radius.
+* `aR_star`/`aRₛ` -- The ratio of the semi-major axis to star radius.
+* `b` -- The impact parameter, bounded between 0 ≤ b ≤ 1.
+* `ecc` -- The eccentricity of the closed orbit, bounded between 0 ≤ ecc < 1.
+* `cos_omega`/`cos_ω` -- The cosine of the argument of periapsis.
+* `sin_omega`/`sin_ω` -- The sine of the argument of periapsis.
+* `incl` -- The inclination of the orbital plane relative to the axis perpendicular to the
+           reference plane [rad]
+* `omega`/`ω` -- The argument of periapsis [rad].
+* `Omega`/`Ω` -- The longitude of the ascending node [rad].
+* `M_planet`/`Mₚ` -- The mass of the planet [M⊙].
+* `M_star`/`Mₛ` -- The mass of the star [M⊙].
+
+# Valid combinations
+The following flowchart can be used to determine which parameters can define a `KeplerianOrbit`:
+1. The `period` or `a` must be given. If both given, then neither `M_star` or `rho_star` can be defined because the stellar density is now implied.
+1. Only `incl` or `b` can be given.
+1. If `ecc` is given, then `omega` must also be given.
+1. If no stellar parameters are given, the central body is assumed to be the Sun. If only `rho_star` is given, then `R_star` is defined to be 1 solar radius. Otherwise, at most two of `M_star`, `R_star`, and `rho_star` can be given.
+1. Either `t_0` or `t_p` must be given, but not both.
```

Comment:
```suggestion
1. The `period` or `a` must be given. If both given, then neither `M_star` or `rho_star` can be defined because the stellar density is now implied.
2. Only `incl` or `b` can be given.
3. If `ecc` is given, then `omega` must also be given.
4. If no stellar parameters are given, the central body is assumed to be the Sun. If only `rho_star` is given, then `R_star` is defined to be 1 solar radius. Otherwise, at most two of `M_star`, `R_star`, and `rho_star` can be given.
5. Either `t_0` or `t_p` must be given, but not both.
```